### ISSUE TITLE
getBackendSrv().datasourceRequest() depreciated

### ISSUE THREAD

**Issue by jacobsiegel1998** [grafana/tutorials#137]
In the `Get data from an external API` section of `Build a Data Source Plugin`, the getBackendSrv().datasourceRequest() method is depreciated and it is recommended to use fetch() instead. Is this correct? Now when I use fetch I get issues with CORS.

**Comment by jorge0136** [comment]
I am seeing this same issue,`getBackendSrv().datasourceRequest()` appears to be [deprecated](https://github.com/grafana/grafana/blob/9008ebd27a536cd48c30644bcb045bec44603907/packages/grafana-runtime/src/services/backendSrv.ts#L141-L151), suggesting `fetch` be used directly instead. 

As far I can tell it appears that we should be using a [data source proxy](https://grafana.com/docs/grafana/latest/developers/plugins/add-authentication-for-data-source-plugins/#authenticate-using-the-data-source-proxy) to handle CORS issues. There are a few issues with this approach:

* The [documentation](https://grafana.com/docs/grafana/latest/developers/plugins/add-authentication-for-data-source-plugins/#add-a-proxy-route-to-your-plugin) for the data-source proxy shows ` getBackendSrv().datasourceRequest()` as the intended external call interface. 
* It's unclear how to debug proxy'd requests. Is there a logging level with grafana itself that will render those calls to the logs? I have experimented a bit with trying to figure this out but haven't found anything effective yet. 

Other data-source plugins I have found in an attempt to find a pattern to follow: 
* [anodot data-source](https://github.com/anodot/grafana-datasource) 
* [github data-source](https://github.com/grafana/github-datasource/tree/main/src) 
* [Cassandra data-source ](https://github.com/HadesArchitect/GrafanaCassandraDatasource)

The best example I have found that has routes seems to be [scalyr.](https://github.com/scalyr/scalyr-grafana-datasource-plugin/blob/master/src/datasource.js) However they aren't using `fetch`. 

A bit more detail in the relevant [tutorial](https://github.com/grafana/tutorials/blob/master/content/tutorials/build-a-data-source-plugin.md#use-the-property) would be very helpful. With some clarification on the intended use I am happy to update the tutorial.

**Comment by jorge0136** [comment]
Looking at the PR that was merged adding the [deprecation warning](https://github.com/grafana/grafana/pull/25746) brings some useful context.

**Comment by jacobsiegel1998** [comment]
Turns out the CORS issue was a problem on my end, I didn't set up the proxy correctly. I am having no issue migrating from `.datasourceRequest()` to `.fetch()`. However, the docs should probably be updated to reflect this depreciation.

As for the logging, I have seen error logging be returned from the proxy but if you're looking for more in-depth logging I'm also unsure how to do that.

**Comment by jacobsiegel1998** [comment]
Not sure if this is what you were referring to @jorge0136, but I am not able to get console.log() to show up in the console.

**Comment by snonux** [comment]
Agreed, it would be very helpful to get https://grafana.com/tutorials/build-a-data-source-plugin/ updated to reflect the change to the new `fetch()` API. All I can do is trying to copy the new approach from other data source plugins.
